Content.Shared/Preferences/HumanoidCharacterProfile.cs

```diff
@@ -661,6 +684,10 @@ public List<ProtoId<TraitPrototype>> GetValidTraits(IEnumerable<ProtoId<TraitPro
                     continue;
                 }
 
+                // If we don't concern ourselves with groups, we stop here.
+                if (!evaluateGroups)
+                    continue;
```

Comment:
Isn't this code now just going to remove all values that belong to a group, even if they are valid?

The solution to this code mess is to stop calling `GetValidTraits()` a second time in the first place.